### ISSUE TITLE
Doesn't cater for automatic scans

### ISSUE THREAD

**Issue by pzi** [alexjurkiewicz/ecr-scan-image#3]
Hi Alex,

Still working on resolving issues we have while using this action.

The latest is it doesn't seem to cater for the fact that automatic vulnerability scans can be enabled on ECR.
Meaning, when we push a (large) image to ECR and have automatic scans enabled, the action fails with `Cannot read property 'findingSeverityCounts' of undefined` because finding returns with

```js
findings: {
  ...
  imageScanStatus: {
    status: "IN_PROGRESS"
  }
}
```

And it will never enter the polling stage because `findings` are not `null`.

I am in the process of changing the code slightly to check if the scan is already in progress by the time it checks if it needs to poll or not. Will let you know how I go shortly.

Furthermore, we try to give the ECR IAM user the least amount of permissions and therefore don't want to allow `startImageScan`. My suggestion (and what I am currently working on) is to add an extra backwards compatible flag to enable opting out of starting the image scan.

Let me know your thoughts on the above and I am happy to PR changes.

Cheers
Patrik

**Comment by alexjurkiewicz** [comment]
Good pickup.

> I am in the process of changing the code slightly to check if the scan is already in progress by the time it checks if it needs to poll or not. Will let you know how I go shortly.

This sounds good. If a scan is already initiated and in progress, it makes sense to block until the scan completes.

> Furthermore, we try to give the ECR IAM user the least amount of permissions and therefore don't want to allow startImageScan. My suggestion (and what I am currently working on) is to add an extra backwards compatible flag to enable opting out of starting the image scan.

If a scan doesn't exist and this action doesn't have permission to start a scan, I think it would have to fail. But the error message could be "No scan findings found" rather than "AWS API access denied". So this change sounds good. If you were thinking of changing this action to silently succeed, it sounds much more dangerous to me and I'm not sure I agree without understanding a real world use case.

**Comment by alexjurkiewicz** [comment]
Actually, ECR image scanning doesn't have any cost associated with it, so to me it doesn't make much sense to remove this permission from the AWS user. What sort of safety does removing this permission give you? I'm not sure it makes sense to support this use-case with a parameter. That complicates the UI for all users, even if the parameter is optional.

**Comment by neilferreira** [comment]
@alexjurkiewicz You are correct - the IAM user can’t achieve anything by performing a scan, and in addition, it is free, so no harm done if these credentials were ever compromised (we would have bigger problems on our hand, in this instance).

Our logic was more: The scans are already automatic, no need to give the IAM user permission to perform a scan.

I think in short, we only need to cater for automatic scans, and if the scans weren't kicked off automatically, we can safely assume that the user has access to initiate a scan.

**Comment by neilferreira** [comment]
Somewhat related note/TODO: It might also be  worth documenting the IAM requirements for this pipeline to be able to run, we have:

```
resource "aws_iam_user" "ecr_user" {
  name = "${terraform.workspace}-ecr-user"
}

resource "aws_iam_user_policy" "ecr_user_role" {
  name   = "${terraform.workspace}-ecr-user-role"
  user   = aws_iam_user.ecr_user.name
  policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
          "ecr:GetAuthorizationToken"
      ],
      "Effect": "Allow",
      "Resource":"*"
    },
    {
      "Action": [
        "ecr:PutImage",
        "ecr:InitiateLayerUpload",
        "ecr:UploadLayerPart",
        "ecr:CompleteLayerUpload",
        "ecr:BatchCheckLayerAvailability",
        "ecr:DescribeImageScanFindings"
      ],
      "Effect": "Allow",
      "Resource":"${aws_ecr_repository.repo.arn}"
    }
  ]
}
EOF
}
```

**Comment by pzi** [comment]
>If a scan doesn't exist and this action doesn't have permission to start a scan, I think it would have to fail. But the error message could be "No scan findings found" rather than "AWS API access denied". So this change sounds good.

Agreed.

>If you were thinking of changing this action to silently succeed, it sounds much more dangerous to me and I'm not sure I agree without understanding a real world use case.

Definitely should fail.

**Comment by alexjurkiewicz** [comment]
A PR documenting the minimum required IAM permissions required in the README. As would a terraform file in `contrib/` 👼

**Comment by alexjurkiewicz** [comment]
Thanks for your code for this, should be fixed in [v1.4.0](https://github.com/alexjurkiewicz/ecr-scan-image/releases/tag/v1.4.0)
